File: hospital-appointment-backend/src/main/java/com/hospital/appointment/entity/Notification.java
package com.hospital.appointment.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 通知消息实体类
 * 保留原有结构
 */
@Data
@TableName("notification")
public class Notification implements Serializable {
    
    @TableId(type = IdType.INPUT)
    private String notificationID;
    
    /** 患者ID */
    private Long patientID;
    
    /** 医生ID */
    private Long doctorID;
    
    /** 管理员ID */
    private Integer adminID;
    
    /** 通知类型：1-预约成功 2-就诊提醒 3-候补成功 4-退号通知 5-医生回复 6-系统通知 */
    private Integer notificationType;
    
    /** 通知标题 */
    private String title;
    
    /** 通知内容 */
    private String notificationContent;
    
    /** 关联ID */
    private String relatedID;
    
    /** 状态：1-未读 2-已读 */
    private Integer notificationStatus;
    
    /** 发送时间 */
    private LocalDateTime sendTime;
    
    /** 阅读时间 */
    private LocalDateTime readTime;
    
    /** 过期时间 */
    private LocalDateTime expireTime;
}

File: hospital-appointment-frontend/src/api/hospital/notification.ts
/**
 * 通知消息 API
 */

import request from '../request'

export interface Notification {
  notificationID: string
  patientID?: number
  doctorID?: number
  adminID?: number
  notificationType: number
  title: string
  notificationContent: string
  relatedID?: string
  notificationStatus: number
  sendTime: string
  readTime?: string
  expireTime?: string
}

export const notificationAPI = {
  getAll: () =>
    request.get<Notification[]>('/hospital/notification/all'),

  getLatest: (limit = 5) =>
    request.get<Notification[]>('/hospital/notification/latest', { params: { limit } }),

  getByFilter: (patientId?: number, type?: number) =>
    request.get<Notification[]>('/hospital/notification/filter', { params: { patientId, type } }),

  getById: (id: string) =>
    request.get<Notification>(`/hospital/notification/${id}`),

  publish: (data: Partial<Notification>) =>
    request.post<string>('/hospital/notification/publish', data),

  markRead: (id: string) =>
    request.put<string>(`/hospital/notification/${id}/read`),

  delete: (id: string) =>
    request.delete<string>(`/hospital/notification/${id}`)
}
